sip/sip-006-runtime-cost-assessment.md

```diff
@@ -0,0 +1,570 @@
+# SIP 006 Clarity Execution Cost Assessment
+
+## Preamble
+
+Title: Clarity Execution Cost Assessment
+
+Author: Aaron Blankstein <aaron@blockstack.com>
+
+Status: Draft
+
+Type: Standard
+
+Created: 10/19/2019
+
+License: BSD 2-Clause
+
+# Abstract
+
+This document describes the measured costs and asymptotic costs
+assessed for the execution of Clarity code. This will not specify the
+_constants_ associated with those asymptotic cost functions. Those
+constants will necessarily be measured via benchmark harnesses and
+regression analyses. Furthermore, the _analysis_ cost associated with
+this code will not be covered by this proposal.
+
+# Measurements for Execution Cost
+
+Execution cost of a block of Clarity code is broken into 5 categories:
+
+1. Runtime cost: captures the number of cycles that a single
+   processor would require to process the Clarity block. This is a
+   _unitless_ metric, so it will not correspond directly to cycles,
+   but rather is meant to provide a basis for comparison between
+   different Clarity code blocks.
+2. Data write count: captures the number of independent writes
+   performed on the underlying data store (see SIP-004).
+3. Data read count: captures the number of independent reads
+   performed on the underlying data store.
+4. Data write length: the number of bytes written to the underlying
+   data store.
+5. Data read length: the number of bytes read from the underlying
+   data store.
```

Comment:
Do we also want to consider transaction bandwidth?  i.e. number of bytes required to encode the transaction?

Reply:
Yes, but that's not a Clarity measurement.